src/controllers/engine/controllerengine.cpp

```diff
@@ -887,6 +884,7 @@ int ControllerEngine::beginTimer(int interval, QJSValue timerCallback,
     if (!timerCallback.isCallable()) {
         qWarning() << "Invalid timer callback provided to beginTimer."
                    << "Valid callbacks are strings and functions.";
+        m_pScriptEngine->throwError("Invalid timer callback provided to beginTimer.");
```

Comment:
Can we pop up an error dialog with the exception?

Reply:
This already happens. This method is called from JS, but JS code is executed as an effect of one of the following events:

- A connection event
- A timer event
- Code evaluation

So the exception triggered here is caught by QJSEngine and then the value returned is an error. In all the situations listed above, we check for errors and show the error dialog.

Reply:
Ah, okay. I didn't test this yet because I'm letting my computer analyze my whole library to test #1624 right now. In that case, how about getting rid of the qWarning above?

Reply:
Fair enough :)